src/openms/source/KERNEL/MSChromatogram.cpp

```diff
@@ -415,3 +415,52 @@ void MSChromatogram::clear(bool clear_meta_data)
     integer_data_arrays_.clear();
   }
 }
+
+
+OpenMS::MSChromatogram::Iterator set_sum_similar_union(OpenMS::MSChromatogram::Iterator first1,
+                    OpenMS::MSChromatogram::Iterator last1,
+                    OpenMS::MSChromatogram::Iterator first2,
+                    OpenMS::MSChromatogram::Iterator last2,
+                    OpenMS::MSChromatogram::Iterator d_first)
+{
+    for (; first1 != last1; ++d_first) {
+        if (first2 == last2)
+            return std::copy(first1, last1, d_first);
+        if ( round( ( first1->getRT() ) * 1000) == round( (first2->getRT()  * 1000 ) ) )
```

Comment:
can you document a bit what is going on here?

Reply:
I've added some documentation for the function. Let me know if there is more that I should add.